app/src/main/kotlin/com/wire/android/ui/debugscreen/PersistentWebSocketService.kt

```diff
@@ -0,0 +1,164 @@
+package com.wire.android.ui.debugscreen
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.graphics.Color
+import android.os.Build
+import android.os.IBinder
+import androidx.annotation.RequiresApi
+import com.wire.android.BuildConfig
+import com.wire.android.R
+import com.wire.android.di.CurrentSessionFlowService
+import com.wire.android.di.KaliumCoreLogic
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.notification.NotificationConstants.NOTIFICATION_ID
+import com.wire.android.notification.NotificationConstants.WEB_SOCKET_CHANNEL_ID
+import com.wire.android.notification.NotificationConstants.WEB_SOCKET_CHANNEL_NAME
+import com.wire.android.notification.WireNotificationManager
+import com.wire.android.ui.WireActivity
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.sync.ConnectionPolicy
+import com.wire.kalium.logic.feature.session.CurrentSessionFlowUseCase
+import com.wire.kalium.logic.feature.session.CurrentSessionResult
+import com.wire.kalium.logic.functional.fold
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.shareIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class PersistentWebSocketService : Service() {
+
+    @Inject
+    @KaliumCoreLogic
+    lateinit var coreLogic: CoreLogic
+
+    private val job = SupervisorJob()
+    private val scope = CoroutineScope(Dispatchers.IO + job)
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Inject
+    lateinit var notificationManager: WireNotificationManager
+
+    @Inject
+    @CurrentSessionFlowService
+    lateinit var currentSessionFlow: CurrentSessionFlowUseCase
+
+    private val navigationManager: NavigationManager = NavigationManager()
+
+    override fun onBind(intent: Intent?): IBinder? {
+        return null
+    }
+
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        if (intent?.action != null && intent.action.equals(
+                ACTION_STOP_FOREGROUND, ignoreCase = true
+            )
+        ) {
+            stopForeground(true)
+            stopSelf()
+        }
+
+        coreLogic.sessionRepository.currentSession().fold({
+
```

Comment:
I think we could add some `log.e` here, just in case this case happens later on production :D